### DISCUSSION TITLE
awslambda_function_not_publicly_accessible check for V3 and V4

### DISCUSSION THREAD

**Discussion by Deepak-sadere** [prowler-cloud/prowler#5510] [General]
I have observed a significant difference in the output of the awslambda_function_not_publicly_accessible check between Prowler V3 and V4. In my environment, when I run the check:

    V3: Shows 3% failure rate
    V4: Shows a 70% failure rate

Upon further investigation, I reviewed the resource-based policies for several Lambda functions flagged by Prowler V4, and most of them don't seem to have any issues that should mark them as "publicly accessible."

This raises the question of what changes were made in Prowler V4 in identifying the misconfiguration for Lambda functions, and whether this is expected behavior or a potential bug.

**Comment by MrCloudSec** [comment]
Hi @Deepak-sadere, I understand your problem with the AWS Lambda policies, we will review the logic of defining them as public and get back to you!

**Reply by MrCloudSec** [reply]
@Deepak-sadere can you try it again in this PR https://github.com/prowler-cloud/prowler/pull/5593? Thanks.
